src/App/App.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useEffect } from "react";
+import "./App.css";
+import { getBookList } from "../apiCalls";
+import BookList from "../BookList/BookList";
+import { Route, NavLink } from "react-router-dom";
+import { SearchBar } from "../SearchBar/SearchBar";
+import FullBook from "../FullBook/FullBook";
+import { WelcomePage } from "../WelcomePage/WelcomePage";
+import { SearchResults } from "../SearchResults/SearchResults";
+
+const App = () => {
+  const [bookList, setBookList] = useState({});
+  const [userInput, setUserInput] = useState("");
+  const [userSearchResults, setUserSearchResults] = useState([]);
+
+  useEffect(() => {
+    getBookList().then((response) => {
+      const acceptableFormats = response.results.filter((item) => {
+        if (item.formats["text/html"] !== undefined) {
+          return item;
+        }
+      });
+      setBookList(acceptableFormats);
+    });
+    // .catch((err) => {
+    //   setErrorMessage(err)
+    //   setErrorStatus(true)
+    // })
+  }, []);
+
+  const handleSearch = (query) => {
+    setUserInput(query);
+  };
+
+  return (
+    <div className="App">
+      <nav>
+        <NavLink to={"/"}>
+          <h1>Read Now</h1>
+        </NavLink>
+        <NavLink to={"/page/1"}>
+          <button>Top Free Books</button>
+        </NavLink>
+        <SearchBar handleSearch={handleSearch} />
+      </nav>
+
+      <Route exact path="/" render={() => <WelcomePage />} />
+
+      {bookList.length > 0 && (
+        <Route
+          exact
+          path="/book/:book_id"
+          render={(match) => {
+            return (
+              <FullBook
+                bookList={userInput ? userSearchResults : bookList}
```

Comment:
Nice ternary! 😎

Reply:
Thanks.  This one solved a lot of headache!